File: RepoTransBench/target_projects/Java/Go/orhanobut_tracklytics/tests/original/trackable_attribute_annotation_test.go
package original

import (
	"reflect"
	"testing"
)

type DummyAttribute struct{}

func (DummyAttribute) AnnotatedMethod(param string) {}

func TestTrackableAttributeOnMethod(t *testing.T) {
	m, ok := reflect.TypeOf(DummyAttribute{}).MethodByName("AnnotatedMethod")
	if !ok {
		t.Fatalf("Method AnnotatedMethod not found")
	}
	// No annotation in Go, but method exists
	if m.Type.Kind() != reflect.Func {
		t.Fatalf("Expected a function type for AnnotatedMethod")
	}
}

func TestTrackableAttributeOnParameter(t *testing.T) {
	m, ok := reflect.TypeOf(DummyAttribute{}).MethodByName("AnnotatedMethod")
	if !ok {
		t.Fatalf("Method AnnotatedMethod not found")
	}
	// Go does not have parameter-level annotations, only for demonstration.
	if m.Type.NumIn() != 2 {
		t.Fatalf("Expected two parameters (receiver + param), got %d", m.Type.NumIn())
	}
}

func TestTargetTypeOnAnnotation(t *testing.T) {
	m, ok := reflect.TypeOf(DummyAttribute{}).MethodByName("AnnotatedMethod")
	if !ok {
		t.Fatalf("Method AnnotatedMethod not found")
	}
	if m.Type.Kind() != reflect.Func {
		t.Fatalf("Expected function kind")
	}
}

func TestRetentionPolicy(t *testing.T) {
	// No direct annotation retention reflection, so this is a placeholder for concept
	if false {
		t.Fatalf("Should be at runtime in Java")
	}
}

File: RepoTransBench/target_projects/Java/C#/orhanobut_tracklytics/tests/original/TrackableAttributeAnnotationTest.cs
using System;
using System.Linq;
using System.Reflection;
using Xunit;

namespace Orhanobut.Tracklytics.Tests
{
    public class TrackableAttributeAnnotationTest
    {
        class Dummy
        {
            [TrackableAttribute]
            public void AnnotatedMethod([TrackableAttribute] string param) { }
        }

        [Fact]
        public void TestTrackableAttributeOnMethod()
        {
            MethodInfo m = typeof(Dummy).GetMethod("AnnotatedMethod");
            Assert.True(m.GetCustomAttributes(typeof(TrackableAttribute), false).Any());
        }

        [Fact]
        public void TestTrackableAttributeOnParameter()
        {
            MethodInfo m = typeof(Dummy).GetMethod("AnnotatedMethod");
            var p = m.GetParameters()[0];
            Assert.True(p.GetCustomAttributes(typeof(TrackableAttribute), false).Any());
        }

        [Fact]
        public void TestTargetTypeOnAnnotation()
        {
            var ta = typeof(Dummy).GetMethod("AnnotatedMethod").GetCustomAttribute<TrackableAttribute>();
            Assert.NotNull(ta);
            var targetAttr = typeof(TrackableAttribute).GetCustomAttribute<AttributeUsageAttribute>();
            Assert.NotNull(targetAttr);
            // AttributeTargets.Method + AttributeTargets.Parameter
            ((targetAttr.ValidOn & (AttributeTargets.Method | AttributeTargets.Parameter)).Should().NotBe(0));
        }

        [Fact]
        public void TestRetentionPolicy()
        {
            // .NET attribute retention is always runtime when set as [AttributeUsage(..., Inherited = true)]
            var attr = typeof(TrackableAttribute).GetCustomAttribute<AttributeUsageAttribute>();
            Assert.NotNull(attr);
            attr.Inherited.Should().BeTrue();
        }
    }
}